crates/extensions/tedge_http_ext/src/actor.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{HttpConfig, HttpError, HttpRequest, HttpResponse};
+use async_trait::async_trait;
+use tedge_actors::{Actor, Address, ChannelError, Mailbox, Message, Recipient, Sender};
+
+pub(crate) struct HttpActor {
+    client: reqwest::Client,
+}
+
+impl HttpActor {
+    pub(crate) fn new(_config: HttpConfig) -> Result<Self, HttpError> {
+        let client = reqwest::Client::builder().build()?;
+        Ok(HttpActor { client })
+    }
+}
+
+#[async_trait]
+impl Actor for HttpActor {
+    type Input = (usize, HttpRequest);
+    type Output = (usize, Result<HttpResponse, HttpError>);
+    type Mailbox = Mailbox<Self::Input>;
+    type Peers = Recipient<Self::Output>;
+
+    async fn run(
+        self,
+        mut requests: Self::Mailbox,
+        mut client: Self::Peers,
+    ) -> Result<(), ChannelError> {
+        while let Some((client_id, request)) = requests.next().await {
```

Comment:
I'm assuming that the real implementation would multiplex between requests coming from multiple `mpsc::Receiver`s of different clients abstracted over a single `Mailbox` instead of putting them all into a single mailbox queue(channel).

Reply:
Presumably this single `Mailbox` is fine in this case as this interacts with `KeyedRecipient` and `RecipientVec`, which abstract away the fact this interacts with multiple clients so we avoid the need for `HttpRequest: Clone` and `HttpResponse: Clone`

Reply:
What I had in mind is the kind of multiplexing describe by @jarhodes314. i.e. an actor processing sequentially requests coming from several clients. To process requests in parallel, one needs to spawn a task for each request (up to some maximum concurrency level).

Reply:
The HTTP actor is using a [`ConcurrentServiceMessageBox`](https://github.com/thin-edge/thin-edge.io/pull/1622/files#diff-85d9e5c8a7a180c4ba09343d893d980d879c01f58dd376d6414d477a9b626cefR154) to handle requests concurrently.